### PR TITLE
Feature: enable/disable packet cache

### LINKED ISSUE #11: Implement 'Cache packets before the inspector is opened' option
There is an 'Options' button in the Packets panel.
There is an option waiting to be implemented

Cache packets before the inspector is opened

Packets are cached till the inspector window is opened, so the user can see also packets fired at the very beginning. It should be possible to switch this off (to avoid performance penalties)

Honza

### PR DESCRIPTION
**Pull request by rpl**
This pull request fixes #11:

- add a new ```showInlineDetails```, ```packetCacheEnabled``` preferences (```false``` by default)
- ```showInlineDetails```, ```packetCacheEnabled``` preferences can be toggled from the addon settings, start button and packets toolbar options menu

#### addon preferences

rdp_inspector_preferences

#### start button

rdp_inspector_startbutton_options

### packets panel toolbar

rdp_inspector_packets-toolbar_options

### CODE REVIEW

File: lib/start-button.js [hunk 1]
```diff
@@ -9,6 +9,7 @@ module.metadata = {
 // Add-on SDK
 const self = require("sdk/self");
 const options = require("@loader/options");
+const { prefs } = require('sdk/simple-prefs');
```
Comment (janodvarko) [inline lib/start-button.js]: Use quotes, so it's consistent with the other require calls.

Honza

File: lib/start-button.js [hunk 2]
```diff
@@ -100,6 +101,15 @@ var StartButton =
 
     items.push({
       nol10n: true,
+      label: Locale.$STR("rdpInspector.menu.packetCache.label"),
+      tooltiptext: Locale.$STR("rdpInspector.menu.packetCache.tip"),
+      type: "checkbox",
+      checked: prefs.packetCacheEnabled,
+      command: this.onToggleCache.bind(this)
+    })
```
Comment (janodvarko) [inline lib/start-button.js]: We are now having two places in the UI where options can be set (within the console and in the Start Button). I think we should have both options available on both places (inline previews and the cache option). And the Start menu should have "Options" submenu at the top separated from other menu items by a separator.

Honza

File: lib/toolbox-overlay.js [hunk 3]
```diff
@@ -10,6 +10,8 @@ module.metadata = {
 const options = require("@loader/options");
 const { Cu, Ci } = require("chrome");
 const { Class } = require("sdk/core/heritage");
+const simplePrefs = require('sdk/simple-prefs');
```
Comment (janodvarko) [inline lib/toolbox-overlay.js]: Use quotes

File: lib/toolbox-overlay.js [hunk 4]
```diff
@@ -42,6 +44,15 @@ const ToolboxOverlay = Class(
 
     Trace.sysout("ToolboxOverlay.initialize;", options);
 
+    // handler of packetCache toggling
+    this.onToggleCache = () => {
+      this.packetCache = prefs.packetCacheEnabled ? [] : null;
+    };
```
Comment (janodvarko) [inline lib/toolbox-overlay.js]: Please define the method the same way as the other methods (like e.g. onSendPacket)

Honza

File: lib/toolbox-overlay.js [hunk 5]
```diff
@@ -42,6 +44,15 @@ const ToolboxOverlay = Class(
...
+    // set initial value
+    this.onToggleCache();
```
Comment (janodvarko) [inline lib/toolbox-overlay.js]: nit: new line

**Comment by janodvarko** [comment]
Looks good to me!

Please see my comments

Honza

**Comment by janodvarko** [comment]
I am not sure about the name of the option: "Do not cache packets before the inspector is opened".
It's descriptive, but a bit long (rather looks like a text for tooltip).

I'll merge it now, but we can yet discus this later.

Thanks for the patch!

Honza

**Comment by rpl** [comment]
@janodvarko I agree, the option name is too long.

My idea is to turn the two options into "checkbox"-styled menu items (mostly like the options menu in the start button) and use the same strings (so they will be recognized immediately as the same options in different places) in a follow up PR

**Pull request merged**